plugins/legacy-plugin-chart-pivot-table/src/PivotTable.js

```diff
@@ -44,25 +45,37 @@ const propTypes = {
 };
 
 function PivotTable(element, props) {
-  const { data, height, columnFormats, numberFormat, numGroups, verboseMap } = props;
+  const { data, height, columnFormats, dateFormat, numberFormat, numGroups, verboseMap } = props;
 
   const { html, columns } = data;
   const container = element;
   const $container = $(element);
+  const dateFormatter =
+    dateFormat === 'smart_date' ? smartDateFormatter : getTimeFormatter(dateFormat);
```

Comment:
`getTimeFormatter('smart_date')` should work just fine. But since we are here, can you update it to follow the same logic as the table chart? 

https://github.com/apache-superset/superset-ui/blob/d323ce86dd7000e9b65a2a7d6b861d713616701b/plugins/plugin-chart-table/src/transformProps.ts#L138-L151

It has fallbacks to d3-format in pre-defined columns and uses Time Grain for `smart_date`.

Reply:
If the code was hard to follow for you, I can also update this later and refactor these logics into some kind of common utilities.

Reply:
Thanks for this reference! I didn't have strong opinions here aka. I was just out to make this work. I'll digest the table chart logic and see if it makes sense for me to streamline here 👍

Reply:
@ktmud I implemented a slightly modified version of your formatter that felt most suitable for this slightly simpler case. Does it make sense to you?

Reply:
Looks good! Thanks for the update. Do you mind adding a datetime column to [Storybook](https://github.com/apache-superset/superset-ui/blob/d631b17597483f981f5de67d2c482e8fae94dc55/packages/superset-ui-demo/storybook/stories/plugins/legacy-plugin-chart-pivot-table/Stories.tsx#L27) as well?